docscan/session/create/check/requested_check.go

```diff
@@ -1,11 +1,12 @@
 package check
 
-// RequestedCheck requests creation of a Check to be performed on a document
-type RequestedCheck struct {
-	Type   string               `json:"type"`
-	Config RequestedCheckConfig `json:"config"`
+// RequestedChecker requests creation of a Check to be performed on a document
+type RequestedChecker interface {
+	Type() string
+	Config() RequestedCheckConfig
+	MarshalJSON() ([]byte, error)
 }
 
 // RequestedCheckConfig is the configuration applied when creating a Check
-type RequestedCheckConfig struct {
+type RequestedCheckConfig interface {
```

Comment:
I suggest leaving this out, unless you plan on adding some methods. It's not different from `interface{}`

Reply:
I think i did this so that we can add methods to this interface later down the line, without it being a breaking change. 

Also with methods like:
```go
// Config is the configuration of the Requested Check
func (c RequestedDocumentAuthenticityCheck) Config() RequestedCheckConfig {
	return RequestedCheckConfig(
		c.config,
	)
}
```
it's a bit more clear what we expect, rather than:
```go
// Config is the configuration of the Requested Check
func (c RequestedDocumentAuthenticityCheck) Config() interface{} {
	return interface{}(
		c.config,
	)
}
```

Reply:
Yeah, OK. Fair enough